packages/netlify-cms-core/src/components/Editor/EditorPreviewPane/EditorPreviewPane.js

```diff
@@ -87,8 +88,17 @@ export default class PreviewPane extends React.Component {
     }
 
     const labelledWidgets = ['string', 'text', 'number'];
-    if (Object.keys(this.inferedFields).indexOf(name) !== -1) {
-      value = this.inferedFields[name].defaultPreview(value);
+
+    const inferedField = Object.entries(this.inferedFields)
+      .filter(([key]) => {
+        // identifier fields can reference nested (e.g. 'en.title') fields so we get the last part
+        const array = keyToPathArray(key);
+        return name === array[array.length - 1];
+      })
+      .map(([, value]) => value)[0];
```

Comment:
Concerned that this could cause order of properties, which the CMS doesn't allow you to control, to influence field inference. Eg., if "books[0].title" showed up in the frontmatter before "title" for some reason, we'd likely end up with the wrong value for title (which will often double as identifier_field).